packages/api/src/utils/server/parseAcceptHeader.ts

```diff
@@ -0,0 +1,75 @@
+import {ResponseFormat} from "../../interfaces.js";
+
+enum MediaType {
+  json = "application/json",
+  ssz = "application/octet-stream",
+}
+
+export const MEDIA_TYPES: {
+  [K in ResponseFormat]: MediaType;
+} = {
+  json: MediaType.json,
+  ssz: MediaType.ssz,
+};
+
+function responseFormatFromMediaType(mediaType: MediaType): ResponseFormat {
+  for (const k of Object.keys(MEDIA_TYPES) as ResponseFormat[]) {
+    if (MEDIA_TYPES[k] == mediaType) {
+      return k;
+    }
+  }
+
+  return "json";
+}
+
+export function parseAcceptHeader(accept?: string): ResponseFormat {
+  // Use json by default.
+  if (!accept) {
+    return "json";
+  }
+
+  // Respect Quality Values per RFC-9110
+  // Acceptable mime-types are comma separated with optional whitespace
+  return responseFormatFromMediaType(
+    accept
+      .toLowerCase()
+      .split(",")
+      .map((x) => x.trim())
+      .reduce(
+        (best: [number, MediaType], current: string): [number, MediaType] => {
+          // An optional `;` delimeter is used to separate the mime-type from the weight
+          // Normalize here, using 1 as the default qvalue
+          const quality = current.includes(";") ? current.split(";") : [current, "q=1"];
+
+          const mediaType = quality[0].trim() as MediaType;
+
+          // If the mime type isn't acceptable, move on to the next entry
+          if (!Object.values(MediaType).includes(mediaType)) {
```

Comment:
Perhaps we can move the `Object.values` call outside of the loop so it is not run with each iteration.  How does on line 30 after the conditional feel?